lectures/functions.md

```diff
@@ -359,49 +359,33 @@ function*---as we did above.
 Recall that $n!$ is read as "$n$ factorial" and defined as
 $n! = n \times (n - 1) \times \cdots \times 2 \times 1$.
 
+We will only consider n as a positive integer here.
+
 There are functions to compute this in various modules, but let's
 write our own version as an exercise.
 
-In particular, write a function `factorial` such that `factorial(n)` returns $n!$
+1. In particular, write a function `factorial` such that `factorial(n)` returns $n!$
 for any positive integer $n$.
-```
-
-```{exercise}
-:label: exercise_2
-
-The [binomial random variable](https://en.wikipedia.org/wiki/Binomial_distribution) $Y \sim Bin(n, p)$ represents the number of successes in $n$ binary trials, where each trial succeeds with probability $p$.
-
-Without any import besides `from numpy.random import uniform`, write a function
-`binomial_rv` such that `binomial_rv(n, p)` generates one draw of $Y$.
-
-Hint: If $U$ is uniform on $(0, 1)$ and $p \in (0,1)$, then the expression `U < p` evaluates to `True` with probability $p$.
-```
-
-```{exercise}
-:label: exercise_3
-
-First, write a function that returns one realization of the following random device
 
-1. Flip an unbiased coin 10 times.
-1. If a head occurs `k` or more times consecutively within this sequence at least once, pay one dollar.
-1. If not, pay nothing.
+2. In addition, try to add a new argument for your function.
+The argument will take in a lambda expression f that transforms n to $f(n) = n^2 + 1$ if n is even, and $f(n) = n^2$ if n is odd. 
```

Comment:
@HumphreyYang , I'm still trying to think how we can maximize clarity here.   Lambda expressions are often annonymous so "take a lambda expression f" might generate some confusion.

Instead of "take in a lambda expression f..." perhaps "take a function `f` that..." and then "Use a lambda expression to define the function `f`."   How does that sound?

If you think my version is excessive then please feel free to ignore.